src/test/kotlin/com/parmet/buf/gradle/AbstractGenerateTest.kt

```diff
@@ -38,4 +38,27 @@ abstract class AbstractGenerateTest : AbstractBufIntegrationTest() {
             listOf("build", "bufbuild", "generated", "java", "com", "google", "type", "DateTime.java")
         assertThat(Paths.get(projectDir.absolutePath, *generatedPathElements.toTypedArray()).toFile().exists()).isTrue()
     }
+
+    @Test
+    fun `buf generate with buf gen template file override`() {
+        gradleRunner().withArguments(BUILD_TASK_NAME).build()
+    }
+
+    @Test
+    fun `buf generate fails with a nonexistent specified template file`() {
+        val result = gradleRunner().withArguments(BUF_GENERATE_TASK_NAME).buildAndFail()
+        assertThat(result.output).contains("No buf.gen.yaml file found in the root directory or with templateFileLocation")
```

Comment:
Perhaps this error message should be specific since in theory the user specified a template - "Specified templateFileLocation does not exist"

Reply:
Also, what happens if you specify a template file that doesn't exist and a default does exist? I think that case should get into tests as well.

Reply:
It would make sense to me to warn if we fallback on the default one and continue. I'll add a test case for that.

Reply:
Hmm. I think it should fail-fast and error in that case - the plugin isn't really respecting the user's intent if they specify a file and it uses a different one.

Reply:
Sounds good, updated the logic and now treating that scenario as a failure case.